gitlab-ci/check-wpscan-v2.php

```diff
@@ -0,0 +1,186 @@
+                    'closed' => false
+                ];
+            }
+        }
+    }
+
+    return $items;
+}
+
+function get_plugin_data($pluginPath) {
+    $pluginFile = $pluginPath . '/' . basename($pluginPath) . '.php';
+    return file_exists($pluginFile) ? get_header_data($pluginFile, 'plugin') : null;
+}
+
+function get_theme_data($themePath) {
+    $themeFile = $themePath . '/style.css';
+    return file_exists($themeFile) ? get_header_data($themeFile, 'theme') : null;
+}
+
+function get_header_data($file, $type) {
+    $fileData = file_get_contents($file);
+    $regex = $type === 'plugin' ? '/^\s*\*\s*Version:\s*(.*)$/m' : '/^Version:\s*(.*)$/m';
+    return preg_match($regex, $fileData, $matches) ? trim($matches[1]) : null;
+}
+
+function get_vulns($extension_slug, $extension_type) {
+    $api_type_path = $extension_type === 'plugin' ? 'plugins' : 'themes';
+    $api_url = "https://wpscan.com/api/v3/$api_type_path/{$extension_slug}";
+    return make_wpscan_api_request($api_url);
+}
+
+function make_wpscan_api_request($api_url) {
+    $token = getenv('WPSCAN_API');
+    $options = ['http' => ['method' => "GET", 'header' => "Authorization: Token token=$token\r\n"]];
+    $context = stream_context_create($options);
+    $result = file_get_contents($api_url, false, $context);
+
+    if ($result === false) return [];
+
+    $parsed = json_decode($result, true);
+    $response = [];
+
+    if (!empty($parsed) && isset(array_values($parsed)[0]['vulnerabilities'])) {
+        foreach (array_values($parsed)[0]['vulnerabilities'] as $vulnerability) {
+            $response[] = [
+                'id' => $vulnerability['id'] ?? null,
+                'title' => $vulnerability['title'] ?? null,
+                'created_at' => $vulnerability['created_at'] ?? null,
+                'updated_at' => $vulnerability['updated_at'] ?? null,
+                'published_date' => $vulnerability['published_date'] ?? null,
+                'vuln_type' => $vulnerability['vuln_type'] ?? null,
+                'references' => $vulnerability['references'] ?? null,
+                'verified' => $vulnerability['verified'] ?? false,
+                'poc' => $vulnerability['poc'] ?? null,
+                'description' => trim(str_replace("\r\n", " ", $vulnerability['description'] ?? '')),
+                'cvss' => [
+                    'score' => $vulnerability['cvss']['score'] ?? null,
+                    'vector' => $vulnerability['cvss']['vector'] ?? null,
+                    'severity' => $vulnerability['cvss']['severity'] ?? null
+                ],
+                'fixed_in' => $vulnerability['fixed_in'] ?? null,
+                'introduced_in' => $vulnerability['introduced_in'] ?? null,
+                'old_id' => $vulnerability['old_id'] ?? null,
+                'metasploit' => $vulnerability['metasploit'] ?? null
+            ];
+        }
+    }
+
+    return $response;
+}
+
+function initialize_security_checks() {
```

Comment:
Can you elaborate what role this function plays? It appears that all it does is return a static array that is then part of the `$full_output` and therefore potentially submitted to the webhook URL, if available.

Since it is static data, I am not sure if there is much value in it?

Reply:
Hi @ArSn the security_checks() function provides the categories of checks the scanner performed. Even though it’s static, downstream systems receiving the webhook expect this information to know what types of checks were run during the scan. TLDR its a piece of metadata that downstream webhook is expecting, without it, it breaks the ingest.

Reply:
Thanks for the feedback! Well, that is not overly general then, but examples can always be adjusted to everyone's liking, of course. 🙂 

Thanks again for contributing this!